subiquity/server/controllers/identity.py

```diff
@@ -135,3 +158,49 @@ async def validate_username_GET(self, username: str) -> UsernameValidation:
             return UsernameValidation.TOO_LONG
 
         return UsernameValidation.OK
+
+    # The identity section (and user-data) section are entirely optional
+    # on Desktop, but setting the variant doesn't happen until controllers
+    # are started. For the identity controller to properly validate
+    # this behavior, we need to do look-ahead on the source controller
+    # logic for loading the available sources.
+    # See: https://github.com/canonical/subiquity/pull/1965
+    def _install_source_is_desktop(self) -> bool:
```

Comment:
similar feedback here, I'd like to lift this to a utility function in `subiquity/server/controllers/identity.py`